### PR TITLE
Handle notifications for swap txs

### LINKED ISSUE #778: Handle swap in notifications
We need to match the behavior to breez mobile.
Once a swap notification is received we need to:
1. Explicitly try to redeem the swap funds (complete swap)
2. Open the app.

### PR DESCRIPTION
**Pull request by ok300**
Fixes #778

### CODE REVIEW

**Comment by ok300** [comment]
> Explicitly try to redeem the swap funds (complete swap)
> #778

The most direct way to do that is to call `breez_sdk_core::swap_in::swap::redeem_swap(bitcoin_address)`. However that's a private method.

The way I found to get to it is to call `breezSDK.inProgressSwap()`. This internally calls `execute_pending_swaps()` which eventuall calls `redeem_swap()`.

@roeierez what do you think?

**Comment by roeierez** [comment]
> > Explicitly try to redeem the swap funds (complete swap)
> > #778
> 
> The most direct way to do that is to call `breez_sdk_core::swap_in::swap::redeem_swap(bitcoin_address)`. However that's a private method.
> 
> The way I found to get to it is to call `breezSDK.inProgressSwap()`. This internally calls `execute_pending_swaps()` which eventuall calls `redeem_swap()`.
> 
> @roeierez what do you think?

I think we should expose explicitly `redeem_swap` as notifications may be given on old swaps (not in progress).

File: android/app/src/main/kotlin/com/cBreez/client/job/RedeemSwap.kt [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+    private val context: Context,
+    private val fgService: ForegroundService,
+    private val payload: String,
+) : SDKJob {
+    companion object {
+        private const val TAG = "RedeemSwapJob"
+    }
+
+    override fun start(breezSDK: BlockingBreezServices) {
+        val request = Json.decodeFromString(AddressTxsConfirmedRequest.serializer(), payload)
```
Comment (erdemyerebasmaz) [inline android/app/src/main/kotlin/com/cBreez/client/job/RedeemSwap.kt:32]: Is thing going to be used with `redeemSwap` API?

**Review by erdemyerebasmaz — COMMENTED** [review]
Changes look structurally good!

Have one comment about an unused variable, which I presume will be used with `redeemSwap` once it's exposed through SDK.

**Comment by erdemyerebasmaz** [comment]
Rebased branch on top of `main` branch to fix CI issues.
- Flutter 3.19 changes
- Breaking SDK change: Use `ConnectRequest` param on `connect` API 
- _Optional SDK branch selection on CI/CD workflow_

**Comment by ok300** [comment]
The PR is ready for final review.

**Review by dangeross — APPROVED** [review]
LGTM

File: android/app/src/main/kotlin/com/cBreez/client/job/RedeemSwap.kt [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+
+        try {
+            breezSDK.rescanSwaps()
```
Comment (roeierez) [inline android/app/src/main/kotlin/com/cBreez/client/job/RedeemSwap.kt:33]: This is a very heavy call as it goes over all historical addresses and I think we don't want to use it within the job that is resources and time limited.
We need to find a way to refresh this specific address information.

**Review by roeierez — CHANGES_REQUESTED** [review]

File: android/app/src/main/kotlin/com/cBreez/client/job/RedeemSwap.kt (hunk 2 above)
Comment (ok300) [reply android/app/src/main/kotlin/com/cBreez/client/job/RedeemSwap.kt:33]: Would this be a good solution? breez/breez-sdk/pull/823

Comment (ok300) [reply android/app/src/main/kotlin/com/cBreez/client/job/RedeemSwap.kt]: Call to `rescanSwaps` removed: e05f412412efe6cde7df5c3d6411861489dddf3c

**Review by roeierez — APPROVED** [review]

**Pull request merged**